charms/feast-integrator/lib/charms/feast_integrator/v0/feast_store_configuration.py

```diff
@@ -0,0 +1,333 @@
+
+        # Update relation data
+        logger.info(f"Sending data {relation_data}")
+        relation.data[self.charm.app].update(relation_data)
+
+
+class FeastStoreConfigurationRequirer(Object):
+    """Implement the Requirer end of the Feast Configuration relation.
+
+    Attributes:
+        charm (CharmBase): the requirer application
+        relation_name (str, optional): the name of the relation
+    """
+
+    def __init__(self, charm, relation_name: Optional[str] = DEFAULT_RELATION_NAME):
+        super().__init__(charm, relation_name)
+        self.relation_name = relation_name
+
+    def get_feature_store_yaml(self):
+        """Generate the Feast feature_store.yaml content from a FeastConfiguration instance.
+
+        Args:
+            config (FeastConfiguration): The configuration values to populate the YAML.
+
+        Returns:
+            str: A string representation of the feature_store.yaml file.
+
+        Raises:
+            StoreConfigurationRelationDataMissingError if data is missing
+            StoreConfigurationRelationMissingError: if there is no related application
+        """
+        relation = self.model.get_relation(self.relation_name)
+
+        if not relation:
+            raise FeastStoreConfigurationRelationMissingError()
+
+        relation_data = relation.data[relation.app]
+
+        if not relation_data:
+            raise FeastStoreConfigurationRelationDataMissingError()
+
+        config = FeastStoreConfiguration(**relation_data)
+
+        yaml_dict: Dict = {
+            "project": "feast_project",
+            "registry": {
+                "registry_type": "sql",
+                "path": (
+                    f"postgresql://{config.registry_user}:{config.registry_password}"
+                    f"@{config.registry_host}:{config.registry_port}/{config.registry_database}"
+                ),
+                "cache_ttl_seconds": 60,
+                "sqlalchemy_config_kwargs": {
+                    "echo": False,
+                    "pool_pre_ping": True,
+                },
+            },
+            "provider": "local",
+            "offline_store": {
+                "type": "postgres",
+                "host": config.offline_store_host,
+                "port": int(config.offline_store_port),
+                "database": config.offline_store_database,
+                "db_schema": "public",
+                "user": config.offline_store_user,
+                "password": config.offline_store_password,
+            },
+            "online_store": {
+                "type": "postgres",
+                "host": config.online_store_host,
+                "port": int(config.online_store_port),
+                "database": config.online_store_database,
+                "db_schema": "public",
+                "user": config.online_store_user,
+                "password": config.online_store_password,
+            },
+            "entity_key_serialization_version": 2,
+        }
```

Comment:
this is duplicating the structure in the Secret [template](https://github.com/canonical/feast-operators/blob/main/charms/feast-integrator/src/templates/feature_store_secret.yaml.j2), it would make more sense to centralize the structure. The question here is: where should the structure be defined?
* in the charm lib
* in the charm code
* in a template owned by the charm

Reply:
thanks for the feedback @misohu, I've created https://github.com/canonical/feast-operators/issues/13 to track this